cli/src/test-utils/test-runner.ts

```diff
@@ -13,472 +13,472 @@ import {
 } from "../../../lib/src/models/definitions";
 import { TypeNode } from "../../../lib/src/models/nodes";
 import { valueFromDataExpression } from "../../../lib/src/utilities/data-expression-utils";
+import { TestConfig } from "./common";
 import { TestLogger } from "./test-logger";
 import { TestTimer } from "./test-timer";
 
-/**
- * Run the contract test suite for a contract.
- *
- * @param definition contract definition
- * @param baseStateUrl base state change URL
- * @param baseUrl base URL
- * @param testFilter optional test filter
- */
-export async function runTest(
-  definition: ContractDefinition,
-  baseStateUrl: string,
-  baseUrl: string,
-  testFilter?: TestFilter
-): Promise<boolean> {
-  const testSuiteStartTime = TestTimer.startTime();
-
-  let allPassed = true;
-
-  for (const endpoint of definition.endpoints) {
-    for (const test of endpoint.tests) {
-      if (testFilter) {
-        if (
-          testFilter.endpoint !== endpoint.name ||
-          (testFilter.test && testFilter.test !== test.name)
-        ) {
-          TestLogger.warn(`Test ${endpoint.name}:${test.name} skipped`);
-          continue;
-        }
-      }
-      const testStartTime = TestTimer.startTime();
+export class TestRunner {
+  private readonly config: TestRunnerConfig;
+  private readonly logger: TestLogger;
 
-      TestLogger.log(`Testing ${endpoint.name}:${test.name}`);
-      const correlatedResponse = findCorrelatedResponse(endpoint, test);
-      const result = await executeTest(
-        test,
-        baseStateUrl,
-        baseUrl,
-        endpoint,
-        correlatedResponse,
-        definition.types
-      );
+  constructor(config: TestRunnerConfig) {
+    this.config = config;
+    this.logger = new TestLogger({ debugMode: config.debugMode });
+  }
 
-      if (result) {
-        TestLogger.success(
-          `Test ${endpoint.name}:${test.name} passed (${TestTimer.formattedDiff(
-            testStartTime
-          )})`,
-          { indent: 1 }
-        );
-      } else {
-        TestLogger.error(
-          `Test ${endpoint.name}:${test.name} failed (${TestTimer.formattedDiff(
-            testStartTime
-          )})`,
-          { indent: 1 }
+  /**
+   * Run the contract test suite for a contract.
+   *
+   * @param definition contract definition
+   * @param config test configuration
+   */
+  async test(
+    definition: ContractDefinition,
+    config?: TestConfig
+  ): Promise<boolean> {
+    const testSuiteStartTime = TestTimer.startTime();
+
+    let allPassed = true;
+
+    for (const endpoint of definition.endpoints) {
```

Comment:
just reminding to revisit this for loooops later on

Reply:
👍coverage reports and junit formatted test summary reports will require a bit of a refactor to how this is done. It will be a good time to look at it then.